src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const grouOfKey = item[key];
+
+    if (!result[grouOfKey as any]) {
+      result[grouOfKey as any] = [];
+    }
+
+    result[grouOfKey as any].push(item);
```

Comment:
The use of `any` typecasting for `grouOfKey` circumvents TypeScript's type safety. Consider using a more specific type or handling the case where `item[key]` might not be a string.